ThirdPartyAdapters/vungle/vungle/src/main/java/com/google/ads/mediation/vungle/VungleMediationAdapter.java

```diff
@@ -46,451 +48,308 @@
-                  mMediationRewardedAdCallback =
-                      mMediationAdLoadCallback.onSuccess(VungleMediationAdapter.this);
-                  return;
-                }
-
-                Vungle.loadAd(mPlacement, mAdConfig, VungleMediationAdapter.this);
-              }
-
-              @Override
-              public void onInitializeError(AdError error) {
-                Log.w(TAG, error.getMessage());
-                mMediationAdLoadCallback.onFailure(error);
-                mPlacementsInUse.remove(mPlacement);
-              }
-            });
-  }
-
-  @Override
-  public void showAd(@NonNull Context context) {
-    Vungle.playAd(mPlacement, mAdConfig, VungleMediationAdapter.this);
-  }
-
-  /**
-   * {@link LoadAdCallback} implementation from Vungle.
-   */
-  @Override
-  public void onAdLoad(final String placementId) {
-    if (mMediationAdLoadCallback != null) {
-      mMediationRewardedAdCallback =
-          mMediationAdLoadCallback.onSuccess(VungleMediationAdapter.this);
+    @Override
+    public void initialize(@NonNull Context context,
+                           @NonNull final InitializationCompleteCallback initializationCompleteCallback,
+                           @NonNull List<MediationConfiguration> mediationConfigurations) {
+
+        if (VungleAds.isInitialized()) {
+            initializationCompleteCallback.onInitializationSucceeded();
+            return;
+        }
+        HashSet<String> appIDs = new HashSet<>();
+        for (MediationConfiguration configuration : mediationConfigurations) {
+            Bundle serverParameters = configuration.getServerParameters();
+            String appIDFromServer = serverParameters.getString(KEY_APP_ID);
+
+            if (!TextUtils.isEmpty(appIDFromServer)) {
+                appIDs.add(appIDFromServer);
+            }
+        }
+
+        int count = appIDs.size();
+        if (count <= 0) {
+            AdError error = new AdError(ERROR_INVALID_SERVER_PARAMETERS, "Missing or Invalid App ID.",
+                    ERROR_DOMAIN);
+            Log.w(TAG, error.getMessage());
+            initializationCompleteCallback.onInitializationFailed(error.getMessage());
+            return;
+        }
+
+        String appID = appIDs.iterator().next();
+        if (count > 1) {
+            String logMessage =
+                    String.format(
+                            "Multiple '%s' entries found: %s. Using '%s' to initialize the Vungle SDK.",
+                            KEY_APP_ID, appIDs, appID);
+            Log.w(TAG, logMessage);
+        }
+
+        VungleAds.init(
+                context,
+                appID,
+                new InitializationListener() {
+                    @Override
+                    public void onSuccess() {
+                        initializationCompleteCallback.onInitializationSucceeded();
+                    }
+
+                    @Override
+                    public void onError(@NonNull VungleException e) {
```

Comment:
@jeffreychang-vungle certain sometimes getting a null for the exception which is causing NPE on the (e.getMessage)

Reply:
Not sure if its something we want to catch or not

Reply:
@HuFangshuai  @felixzhang0703 @anton-blashchuk-zoolatech would it be possible to make sure the SDK doesn't send an error as null. It would be a bad developer experience to have to catch that